File: WeChatFlight/WCFlight/src/com/flight/activity/MainDialogFragment.java
package com.flight.activity;

import android.app.AlertDialog;
import android.app.Dialog;
import android.os.Bundle;
import android.support.v4.app.DialogFragment;
import android.view.View;
import android.widget.Button;
import f21.wechat.flight.R;

public class MainDialogFragment extends DialogFragment{
	
	private Button mBtnRestart, mBtnQuit;
	
	@Override
	public Dialog onCreateDialog(Bundle savedInstanceState) {
		View view = getActivity().getLayoutInflater().inflate(R.layout.fragment_main, null);
		
		mBtnRestart = (Button) view.findViewById(R.id.btn_restart);
		mBtnQuit = (Button) view.findViewById(R.id.btn_quit);
		
		AlertDialog.Builder dialog = new AlertDialog.Builder(getActivity());
		dialog.setView(view);
		
		mBtnRestart.setOnClickListener(new View.OnClickListener() {
			
			@Override
			public void onClick(View v) {
				if (getActivity() instanceof AlterInputListener) {
					((AlterInputListener)getActivity()).onAlertInputComplete(true);
					dismiss();
				}
			}
		});
		
		mBtnQuit.setOnClickListener(new View.OnClickListener() {
			
			@Override
			public void onClick(View v) {
				if (getActivity() instanceof AlterInputListener) {
					((AlterInputListener)getActivity()).onAlertInputComplete(false);
					dismiss();
				}
			}
		});
		
		return dialog.create();
	}
	
	public interface AlterInputListener{
		void onAlertInputComplete(boolean isRestart);
	}
}

File: WeChatFlight/Flight/src/main/java/com/flight/activity/MainDialogFragment.kt
package com.flight.activity

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import android.view.View
import android.widget.Button
import androidx.fragment.app.DialogFragment
import com.project.wechatflight.R

class MainDialogFragment : DialogFragment() {
    private var mBtnRestart: Button? = null
    private var mBtnQuit: Button? = null
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val view = activity!!.layoutInflater.inflate(R.layout.fragment_main, null)
        mBtnRestart = view.findViewById<View>(R.id.btn_restart) as Button
        mBtnQuit = view.findViewById<View>(R.id.btn_quit) as Button
        val dialog = AlertDialog.Builder(activity)
        dialog.setView(view)
        mBtnRestart!!.setOnClickListener {
            if (activity is AlterInputListener) {
                (activity as AlterInputListener?)!!.onAlertInputComplete(true)
                dismiss()
            }
        }
        mBtnQuit!!.setOnClickListener {
            if (activity is AlterInputListener) {
                (activity as AlterInputListener?)!!.onAlertInputComplete(false)
                dismiss()
            }
        }
        return dialog.create()
    }

    interface AlterInputListener {
        fun onAlertInputComplete(isRestart: Boolean)
    }
}